### PR TITLE
Bugfix quantity

### LINKED ISSUE #153: [PE-D] Bugs in Add command quantity
Says must be a positive number even though 1.3 is a positive number. Maybe rewording the sentence to say positive integer might be better.

Says must be a positive number even though 111111111111 is a positive number. Should try checking for max value of integer.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: hussain1998/ped#1

### LINKED ISSUE #154: [PE-D] Remove quantity
Typed in a large positive number: 
Expected error: `Not enough in fridge to remove!`
Even though number entered is positive, error shows enter a positive number

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: hussain1998/ped#2

### LINKED ISSUE #157: [PE-D] Bug in Add quantity
Integer overflow occurs when adding quantity to the food.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: hussain1998/ped#5

### LINKED ISSUE #159: [PE-D] fridgeData.txt file
I can edit the quantity in the file to be negative and the programs loads successfully.
Expected an error.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: hussain1998/ped#7

### LINKED ISSUE #161: [PE-D] limitsData.txt file
I can edit the numbers to negative and the program does not give an error.
Expected an error.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: hussain1998/ped#9

### LINKED ISSUE #168: [PE-D] Functionality Bugs
The error message is not exactly accurate. It would be better to change it to positive integer number. 

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Cocokkkk/ped#6

### LINKED ISSUE #191: [PE-D] remove FOODNAME command undesired error
The command doesn't handle the situation correctly when the user types in a negative number for the quantity to be removed. Initially there were 3 apples as shown below. When I typed remove apples /qty -1 , I expected an error that tells me to only input positive values but instead it added another one to item. When I listed, it showed the qty of apples to be 4 which is not the desired outcome. So, maybe you can handle a situation where undesirable parameters are keyed in by the user.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: vaiish371/ped#3

### LINKED ISSUE #197: [PE-D] Removing negative quantity of food
Using the "remove" command with negative quantity would add to the original quantity. Perhaps the team might want to add a check to see if the value is negative.

Step 1: Enter "add chicken /cat meat /exp 30-06-2021 /loc lower_shelf /qty 200"
Step 2: Enter "remove chicken /qty -100"

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: 8kdesign/ped#3

### LINKED ISSUE #198: [PE-D] Value overflow for food quantity
When a ridiculous amount of the same food is added, the quantity value would overflow and become negative.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: 8kdesign/ped#4

### LINKED ISSUE #199: [PE-D] Negative value for category limit
Entering a large negative number for quantity in "setlimit" command would print an error asking for positive value. However, when a smaller negative number is entered, the value is accepted.

Step 1: "setlimit meat /qty -999999"

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: 8kdesign/ped#5

### PR DESCRIPTION
**Pull request by SimJJ96**

### CODE REVIEW

File: src/main/java/seedu/fridgefriend/food/Fridge.java [hunk 1]
```diff
@@ -49,12 +52,15 @@ public void removeByIndex(int index) {
...
-        int deltaQuantity = newFood.getQuantity();
-        int oriQuantity = existingFood.getQuantity();
-        int newQuantity = deltaQuantity + oriQuantity;
-        newFood.setQuantity(newQuantity);
-        existingFood.setQuantity(newQuantity);
+    private void editFoodQuantity(Food newFood, Food existingFood) throws InvalidQuantityException {
+        long deltaQuantity = newFood.getQuantity();
+        long oriQuantity = existingFood.getQuantity();
+        long newQuantity = deltaQuantity + oriQuantity;
+        if (newQuantity >= Integer.MAX_VALUE) {
```
Comment (Leeyp) [inline src/main/java/seedu/fridgefriend/food/Fridge.java]: We might want to set the "MAX Value" not to integer max value, but rather a smaller value (perhaps 10k? 100k?) which would definitely prevent the case of Integer Overflows.

**Review by Leeyp — CHANGES_REQUESTED** [review]
Clarified with author, ignore

**Review by Leeyp — APPROVED** [review]
LGTM

**Review by Vinci-Hu — APPROVED** [review]
LGTM

File: src/main/java/seedu/fridgefriend/food/Fridge.java [hunk 2]
```diff
@@ -49,12 +59,15 @@ public void removeByIndex(int index) {
...
     //@@author Vinci-Hu
-    private void editFoodQuantity(Food newFood, Food existingFood) {
-        int deltaQuantity = newFood.getQuantity();
-        int oriQuantity = existingFood.getQuantity();
-        int newQuantity = deltaQuantity + oriQuantity;
-        newFood.setQuantity(newQuantity);
-        existingFood.setQuantity(newQuantity);
+    private void editFoodQuantity(Food newFood, Food existingFood) throws InvalidQuantityException {
+        long deltaQuantity = newFood.getQuantity();
+        long oriQuantity = existingFood.getQuantity();
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/food/Fridge.java]: can we name this "originalQuantity" instead

**Review by kwokyto — APPROVED** [review]
Otherwise, LGTM

File: src/main/java/seedu/fridgefriend/food/Fridge.java [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
+
+import seedu.fridgefriend.exception.InvalidQuantityException;
 import seedu.fridgefriend.exception.RepetitiveFoodIdentifierException;
 
 
 public class Fridge {
 
-
     private ArrayList<Food> fridge = new ArrayList<>();
+    private String exceedAddMessage = "Sorry my friend, "
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/food/Fridge.java:12]: can we change this and the one below to like "sorry my kawan"

File: src/main/java/seedu/fridgefriend/exception/InvalidQuantityException.java [hunk 4]
```diff
@@ -6,7 +6,7 @@
  */
 public class InvalidQuantityException extends Exception {
     private static final String errorMessage = "Sorry my friend, the quantity "
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/exception/InvalidQuantityException.java:8]: can we change this to "sorry my kawan"
